basic_test.go

```diff
@@ -78,3 +89,264 @@ func TestCapitalize(t *testing.T) {
 	mb.On("Send", expectedMsg).Return(nil)
 	cb.Capitalize(&mockMsg)
 }
+
+//Bus Timings should be tested in three phases
+//1. Pop three closest busstops according to location
+//2. The right format
+//3. HTTP request works
+
+//TestBus tests the format of the response string.
+func TestBus(t *testing.T) {
+	mb := mockBot{}
+	cb := Cinnabot{
+		bot: &mb,
+	}
+
+	BSH := makeHeap(*mockMsg.Location)
+	oldgetBusStops := makeHeap
+	defer func() { makeHeap =oldgetBusStops} ()
+
+	makeHeap = func(loc tgbotapi.Location) (BusStopHeap) {
+		return BSH
+	}
+
+	oldreadall := ioutil.ReadAll
+	defer func() { ioutil.ReadAll = oldreadall} ()
+
+	//Simulates with a default response @ busStop code 19059
+	ioutil.ReadAll = func(r io.Reader) ([]byte, error) {
+		responseText := "{\"odata.metadata\":\"http://datamall2.mytransport.sg/ltaodataservice/$metadata#BusArrivalv2/" +
+			"@Element\",\"BusStopCode\":\"42109\",\"Services\":[{\"ServiceNo\":\"170\",\"Operator\":\"SBST\",\"NextBus" +
+				"\":{\"OriginCode\":\"46239\",\"DestinationCode\":\"01109\",\"EstimatedArrival\":\"2017-12-22T01:38:47" +
+					"+08:00\",\"Latitude\":\"1.4398775\",\"Longitude\":\"103.768522\",\"VisitNumber\":\"1\",\"Load\":\"" +
+						"SEA\",\"Feature\":\"WAB\",\"Type\":\"SD\"},\"NextBus2\":{\"OriginCode\":\"\",\"DestinationCode\"" +
+							":\"\",\"EstimatedArrival\":\"\",\"Latitude\":\"\",\"Longitude\":\"\",\"VisitNumber\":\"\",\"" +
+								"Load\":\"\",\"Feature\":\"\",\"Type\":\"\"},\"NextBus3\":{\"OriginCode\":\"\",\"Destin" +
+									"ationCode\":\"\",\"EstimatedArrival\":\"\",\"Latitude\":\"\",\"Longitude\":\"\",\"V" +
+										"isitNumber\":\"\",\"Load\":\"\",\"Feature\":\"\",\"Type\":\"\"}}]}"
+		return []byte(responseText), nil
+	}
+
+	expectedMsgStr := "Opp Beauty World Ctr\n================\nBus 170 : 25 minutes\n\nOpp Beauty World Ctr\n=======" +
+		"=========\nBus 170 : 25 minutes\n\nOpp Beauty World Ctr\n================\nBus 170 : 25 minutes\n\n"
+	expectedMsg := tgbotapi.NewMessage(999, expectedMsgStr)
+	mb.On("Send", expectedMsg).Return(nil)
+	cb.BusTimings(&mockMsg)
+}
+
+//Test Weather tests format of output
+func TestWeather(t *testing.T) {
+	mb := mockBot{}
+	cb := Cinnabot{
+		bot: &mb,
+	}
+	oldreadall := ioutil.ReadAll
+	defer func() { ioutil.ReadAll = oldreadall} ()
+
+	//Simulates with a default response
+
+	ioutil.ReadAll = func(r io.Reader) ([]byte, error) {
+		responseText := "{\"area_metadata\":[{\"name\":\"Bukit Batok\",\"label_location\":{\"latitude\":1.353,\"" +
```

Comment:
My plan was to emulate a response from the server and then test the format of the response. 
Should I be testing weather at all?

Reply:
I think it may be sufficient to assert that the API returned a 200 code and contains some keywords. Matching exact strings is probably too fragile.

Testing weather is still a good idea.